Permanent/ViewControllers/LegacyPlanning/LegacyPlanningIntroViewController.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  LegacyPlanningViewController.swift
+//  Permanent
+//
+//  Created by Lucian Cerbu on 25.04.2023.
+//
+
+import UIKit
+
+class LegacyPlanningIntroViewController: BaseViewController<LegacyPlanningViewModel> {
+    @IBOutlet weak var setUpTitleLabel: UILabel!
+    @IBOutlet weak var setUpDescriptionLabel: UILabel!
+    @IBOutlet weak var accountLegacyTitleLabel: UILabel!
+    @IBOutlet weak var accountLegacyDescriptionLabel: UILabel!
+    @IBOutlet weak var archiveLegacyTitleLabel: UILabel!
+    @IBOutlet weak var archiveLegacyDescriptionLabel: UILabel!
+    @IBOutlet weak var setUpMyAccountButton: LegacyPlanningSetUpButton!
+    @IBOutlet weak var tellMoreButton: UIButton!
+    @IBOutlet weak var logoWidthConstraint: NSLayoutConstraint!
+    @IBOutlet weak var logoHeightConstraint: NSLayoutConstraint!
+    @IBOutlet weak var scrollView: UIScrollView!
+    
+    var isShowMoreEnabled: Bool = false {
+        didSet {
+            if isShowMoreEnabled {
+                showElements()
+            } else {
+                hideElements()
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        styleNavBar()
+        setupNavBar()
+        setupUI()
+    }
+    
+    private func setupNavBar() {
+        titleLabelSetup()
+        if let viewControllersCount = navigationController?.viewControllers.count, viewControllersCount > 1 {
+            backButtonSetup()
+        }
+        closeButtonSetup()
+    }
+    
+    private func setupUI() {
+        view.backgroundColor = .darkBlue
+        setupTitleLabel()
+        setupDescriptionLabel()
+        
+        tellMoreButton.setTitleColor(.white, for: .normal)
+        tellMoreButton.setTitle("Tell me more about Legacy Planning".localized(), for: .normal)
+        
+        accountLegacyTitleLabel.isHidden = true
+        accountLegacyDescriptionLabel.isHidden = true
+        archiveLegacyTitleLabel.isHidden = true
+        archiveLegacyDescriptionLabel.isHidden = true
+    }
+    
+    private func titleLabelSetup() {
+        let titleLabel = UILabel()
+        titleLabel.text = "Legacy Planning".localized()
+        titleLabel.textColor = .white
+        titleLabel.font = TextFontStyle.style35.font
+        titleLabel.sizeToFit()
+        
+        navigationItem.titleView = titleLabel
+    }
+    
+    private func backButtonSetup() {
```

Comment:
I do have a feeling that we will have this code through the whole legacy planning. Maybe let's move this into a protocol so we can reuse it.

protocol LegacyPlanningNavigation {
func setupNavigationBar()
}

extension LegacyPlanningNavigation: UIViewController {

func setupNavigationBar() {
// the implementation
}
}